src/main/java/io/openraven/magpie/api/MagpieResource.java

```diff
@@ -210,4 +218,118 @@ public JsonNode getDiscoveryMeta() {
   public void setDiscoveryMeta(JsonNode discoveryMeta) {
     this.discoveryMeta = discoveryMeta;
   }
+
+  public static class MagpieResourceBuilder {
+    private final ObjectMapper mapper;
+    private String assetId;
+    private String resourceName;
+    private String resourceId;
+    private String resourceType;
+    private String region;
+    private String projectId;
+    private String accountId;
+    private Instant createdIso;
+    private Instant updatedIso = Instant.now();
+    private String discoverySessionId;
+    private Long maxSizeInBytes = null;
+    private Long sizeInBytes = null;
+    private JsonNode configuration;
+    private JsonNode supplementaryConfiguration;
+    private JsonNode tags;
+    private JsonNode discoveryMeta;
+
+    public MagpieResourceBuilder(ObjectMapper mapper, String assetId) {
+      this.assetId = assetId;
+      this.mapper = mapper;
+
+      this.configuration = mapper.createObjectNode();
+      this.supplementaryConfiguration = mapper.createObjectNode();
+      this.tags = mapper.createObjectNode();
+      this.discoveryMeta = mapper.createObjectNode();
+    }
+
+    public MagpieResourceBuilder withAssetId(String assetId) {
+      this.assetId = assetId;
+      return this;
+    }
```

Comment:
Is this necessary when it takes assetId as required parameter?

Reply:
Removed